routehandlers/handler_test.go

```diff
@@ -568,13 +542,95 @@ var _ = Describe("Handler", func() {
 						JustBeforeEach(func() {
 							afterActualLRP = nil
 						})
+
 						It("logs an error", func() {
 							routeHandler.HandleEvent(logger, models.NewActualLRPInstanceChangedEvent(beforeActualLRP, afterActualLRP, "some-trace-id"))
 							Expect(logger).To(gbytes.Say("nil-actual-lrp"))
 						})
 					})
 				})
 
+				Context("when after state Routable is true", func() {
+					BeforeEach(func() {
+						afterActualLRP.Routable = true
+					})
+
+					It("should add/update the endpoint on the table", func() {
+						Expect(fakeTable.AddEndpointCallCount()).To(Equal(1))
+
+						_, actualLRP := fakeTable.AddEndpointArgsForCall(0)
+						Expect(actualLRP).To(Equal(afterActualLRP))
+					})
+
+					It("should emit whatever the table tells it to emit", func() {
+						Expect(natsEmitter.EmitCallCount()).Should(Equal(1))
+
+						messagesToEmit := natsEmitter.EmitArgsForCall(0)
+						Expect(messagesToEmit).To(Equal(dummyMessagesToEmit))
+					})
+
+					It("sends a 'routes registered' metric", func() {
+						Eventually(counterChan).Should(Receive(Equal(counter{
+							name:  "RoutesRegistered",
+							delta: 2,
+						})))
+					})
+
+					It("sends a 'routes unregistered' metric", func() {
+						Eventually(counterChan).Should(Receive(Equal(counter{
+							name:  "RoutesUnregistered",
+							delta: 0,
+						})))
+					})
+				})
+
+				Context("when after state Routable is false and before state Routable is true", func() {
+					BeforeEach(func() {
+						beforeActualLRP.Routable = true
+						afterActualLRP.Routable = false
+					})
+
+					It("should not add/update the endpoint on the table", func() {
+						Expect(fakeTable.AddEndpointCallCount()).To(Equal(0))
+					})
+
+					It("should remove the endpoint from the table", func() {
+						Expect(fakeTable.RemoveEndpointCallCount()).To(Equal(1))
+
+						_, actualLRP := fakeTable.RemoveEndpointArgsForCall(0)
+						Expect(actualLRP).To(Equal(afterActualLRP))
+					})
+
+					It("should emit whatever the table tells it to emit", func() {
+						Expect(natsEmitter.EmitCallCount()).To(Equal(1))
+
+						messagesToEmit := natsEmitter.EmitArgsForCall(0)
+						Expect(messagesToEmit).To(Equal(dummyMessagesToEmit))
+					})
+				})
+
+				Context("when after state Routable is false and before state Routable is false", func() {
+					BeforeEach(func() {
+						beforeActualLRP.Routable = false
+						afterActualLRP.Routable = false
+					})
+
+					It("should not add/update the endpoint on the table", func() {
+						Expect(fakeTable.AddEndpointCallCount()).To(Equal(0))
+					})
+
+					It("should not remove the endpoint from the table", func() {
+						Expect(fakeTable.RemoveEndpointCallCount()).To(Equal(0))
+					})
+
+					It("should call emit with empty messages", func() {
```

Comment:
Does it send routes un/registered metrics in this case? If so we should test.

Reply:
Before was false and after is false, so delta 0 for both registered and unregistered